static/app/components/searchSyntax/parser.tsx

```diff
@@ -221,16 +221,52 @@ export const filterTypeConfig = {
 
 type FilterTypeConfig = typeof filterTypeConfig;
 
+/**
+ * Object representing an invalid filter state
+ */
+type InvalidFilter = {
+  /**
+   * The message indicating why the filter is invalid
+   */
+  reason: string;
+  /**
+   * In the case where a filter is invalid, we may be expecting a different
+   * type for this filter based on the key. This can be useful to hint to the
+   * user what values they should be providing.
+   *
+   * This may be multiple filter types.
+   */
+  expectedType?: FilterType[];
+};
+
 type FilterMap = {
   [F in keyof FilterTypeConfig]: {
     type: Token.Filter;
+    /**
+     * The filter type being represented
+     */
     filter: F;
-    config: FilterTypeConfig[F];
+    /**
+     * The key of the filter
+     */
     key: KVConverter<FilterTypeConfig[F]['validKeys'][number]>;
+    /**
+     * The value of the filter
+     */
     value: KVConverter<FilterTypeConfig[F]['validValues'][number]>;
+    /**
+     * The operator applied to the filter
+     */
     operator: FilterTypeConfig[F]['validOps'][number];
+    /**
+     * Indicates if the filter has been negated
+     */
     negated: FilterTypeConfig[F]['canNegate'] extends true ? boolean : false;
-    invalidReason: string | null;
+    /**
+     * When a filter is marked as 'invalid' a reason is given. If the filter is
+     * not invalid this will always be null
+     */
+    invalid: InvalidFilter | null;
```

Comment:
Decided it was time for token filter types to have better comments for their structure.